src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,53 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int COUNTER_FIGURE = 5;
+    private final Random random = new Random();
+    private final NumberSupplier numberSupplier = new NumberSupplier(random);
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        Figure figure;
+
+        int randomNumber = random.nextInt(COUNTER_FIGURE);
+        switch (randomNumber) {
+            case 0 :
+                figure = new Circle(
+                        colorSupplier.getRandomColor(),
+                        numberSupplier.generateRandomNumber());
+                break;
```

Comment:
```suggestion
                return new Circle(
                        colorSupplier.getRandomColor(),
                        numberSupplier.generateRandomNumber());
```

Reply:
So we simplify the construction a little, without creating an extra variable.
You can also do even better, move the creation of each shape into a separate private method. And then it will
return getRandomCircle()